src/units/abilities.cpp

```diff
@@ -667,15 +715,27 @@ bool attack_type::get_special_bool(const std::string& special, bool simple_check
 
 	std::vector<special_match> special_tag_matches;
 	std::vector<special_match> special_id_matches;
-	get_special_children(special_tag_matches, special_id_matches, other_attack_->specials_, special);
-	for(const special_match& entry : special_tag_matches) {
-		if ( other_attack_->special_active(*entry.cfg, AFFECT_OTHER, entry.tag_name) ) {
-			return true;
+	if(special_id && special_tags){
+		get_special_children(special_tag_matches, special_id_matches, other_attack_->specials_, special);
+	}
+	else if (special_id && !special_tags){
```

Comment:
The placement of `else if` is inconsistent here compared to earlier in the function. I won't ask you to fix it this time, but in the future please be consistent about placement of `else` and about (lack of) spaces after `if`.